lib/boolean.js

```diff
@@ -23,23 +28,72 @@ internals.Boolean = class extends Any {
             value
         };
 
-        if (typeof value === 'string' &&
-            options.convert) {
+        result.value = (this._inner._truthySet.has(value) ? true
+            : (this._inner._falsySet.has(value) ? false : value));
+
+        result.errors = (typeof result.value === 'boolean') ? null : this.createError('boolean.base', null, state, options);
+        return result;
+    }
+
+    _truthy() {
```

Comment:
Just following the pattern set with `_allow` and `allow` in Any. I can un-split them if you prefer though.